### ISSUE TITLE
Still pending attestation

### ISSUE THREAD

**Issue by TVikg** [opentimestamps/opentimestamps-server#101]
Hello there.

I saw the pending attestation issue was closed but I've found I have the same problem.

Using the browser, when I verify an OTS file with the stamped file, I obtain the green flag of successfully verified file but if I click on the small "i" in the right upper corner, where you can see the details of the operation, there's still a Pending attestation: server [https://alice.btc.calendar.opentimestamps.org](https://opentimestamps.org/#info)

Can you check what's going on?

**Comment by petertodd** [comment]
A pending attestation is totally normal if you recently created the timestamp, as it usually takes a few hours for a Bitcoin transaction to be mined.

When did you create the timestamp?

**Comment by TVikg** [comment]
Hello Peter

this is occurring with an OTS file produced on 13/09/2023 so likely 2 months ago! But also with another one produced today at 2 p.m.

Thanks for your answer

**Comment by petertodd** [comment]
On Thu, Nov 16, 2023 at 01:35:55PM -0800, TVikg wrote:
> I've sent the file via mail, let me know if you've received it.

Thanks! I got it.

The file you sent me is a valid, fully upgraded, ots proof. It has all the BTC
transaction info in it already so it should be validating just fine even
without the OTS calendars. So unless you accidentally sent me the wrong file,
there must be something wrong with your set. What exact OTS client are you
using?

**Comment by TVikg** [comment]
Hi Todd

I used the web interface of the website, nothing special. I'm not using the shell commands, only the website homepage of OpenTimestamps.

**Comment by petertodd** [comment]
On Thu, Nov 16, 2023 at 02:03:22PM -0800, TVikg wrote:
> Hi Todd
> 
> I used the web interface of the website, nothing special. I'm not using the shell commands, only the website homepage of OpenTimestamps.

Huh, weird. Sorry, but without the file being timestamped I can't easily
replicate this on my side. You're welcome to send it to me privately via email;
I won't share it further without your permission, and I'll delete it when we're
done.

**Comment by TVikg** [comment]
yeah I understand but I can't send you because it's a forensic acquisition. I could send you the file I've timestamped today instead. For how long do you suggest to wait ?

**Comment by petertodd** [comment]
On Thu, Nov 16, 2023 at 02:16:58PM -0800, TVikg wrote:
> yeah I understand but I can't send you because it's a forensic acquisition. I could send you the file I've timestamped today instead. For how long do you suggest to wait ?

That's understandable.

Let's wait 24 hours and see if the timestamp from today still has that problem.
Fees are very high right now so timestamp transactions are happening slowly;
the OTS calendars spend approximately the same amount of money every day. So if
fees are higher, timestamp transactions happen less often.

**Comment by TVikg** [comment]
Ok, I'll write you tomorrow after 4 p.m. more than 24 hour the timestamp.

In the meanwhile the timestamp transaction are happening slowly, is the document timestamp valid?

It would be very useful if the timestamp could show the clock time not only the date, like the TSA ones.

**Comment by petertodd** [comment]
On Thu, Nov 16, 2023 at 02:41:59PM -0800, TVikg wrote:
> Ok, I'll write you tomorrow after 4 p.m. more than 24 hour the timestamp.

Great!

> In the meanwhile the timestamp transaction are happening slowly, is the document timestamp valid?

The timestamp you emailed me is valid. So I suspect the issue you're seeing is
just a glitch on the website.

> It would be very useful if the timestamp could show the clock time not only the date, like the TSA ones.

Agreed. The reason why we don't show that by default is because Bitcoin
timestamps could hypothetically be manipulated by miners, at least to a degree.
Rounding them off to the nearest day is a safe default that is good enough for
the vast majority of timestamp applications.

However, that's not good enough for everyone. For more precise timestamps, I
recommend manually checking the actual timestamp of the specific block that
timestamped your document, and then checking that block timestamp against other
sources.

I have a project that re-timestamps Bitcoin blocks with the trusted timestamp
scheme Roughtime: https://github.com/opentimestamps/restamp

The timestamps restamp makes are traceable back to Cloudflare, among others.
And those signatures are in turn timestamped with Bitcoin, allowing them to be
validated in the future even if Cloudflare, etc., later leaks their private
keys.

Finally, on the TODO list is to add trusted timestamping to OTS directly.
